### PR TITLE
Detect unexpected emulator exit and stops waiting for it

### PR DESCRIPTION
**Pull request by teobugslayer**
Fixes https://github.com/NativeScript/nativescript-cli/issues/645

### CODE REVIEW

File: mobile/android/android-emulator-services.ts [hunk 1]
```diff
@@ -23,6 +24,7 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
 	private static TIMEOUT_SECONDS = 120;
 	private static UNABLE_TO_START_EMULATOR_MESSAGE = "Cannot run your app in the native emulator. Increase the timeout of the operation with the --timeout option or try to restart your adb server with 'adb kill-server' command. Alternatively, run the Android Virtual Device manager and increase the allocated RAM for the virtual device.";
 	private static RUNNING_ANDROID_EMULATOR_REGEX = /^(emulator-\d+)\s+device$/;
+	private static EMULATOR_EXITED_MESSAGE = "The native emulator has stopped unexpectedly. Verify its configuration and try again.";
```
Comment (rosen-vladimirov) [inline mobile/android/android-emulator-services.ts]: do we consider Genymotion as native emulator? The message is used for both native and Genymotion emulators

File: mobile/android/android-emulator-services.ts [hunk 2]
```diff
@@ -224,19 +226,26 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
 
 			// have to start new emulator
 			this.$logger.info("Starting Android emulator with image %s", image);
+			let emulatorProc: child_process.ChildProcess = null;
+			let emulatorProcEnded = false;
 			if(this.$options.geny) {
 				//player is part of Genymotion, it should be part of the PATH.
-				this.$childProcess.spawn("player", ["--vm-name", image],
-					{ stdio: "ignore", detached: true }).unref();
+				emulatorProc = this.$childProcess.spawn("player", ["--vm-name", image],
```
Comment (rosen-vladimirov) [inline mobile/android/android-emulator-services.ts]: this code is holding the terminal, in AppBuilder CLI we want to release the terminal when the emulator is successfully started, that's why we were using `.unref()`

**Comment by rosen-vladimirov** [comment]
The code is breaking the behavior of AppBuilder CLI where the terminal should be released when the simulator starts.

File: mobile/android/android-emulator-services.ts (hunk 1 above)
Comment (teobugslayer) [inline mobile/android/android-emulator-services.ts]: I've modeled the message after UNABLE_TO_START_EMULATOR_MESSAGE - should we change it as well?

Comment (rosen-vladimirov) [inline mobile/android/android-emulator-services.ts]: I suggest to remove "native"  word :)

**Comment by rosen-vladimirov** [comment]
Ignore my comment, there's another issue in AppBuilder CLI, which is causing emulator hold the console. I'll fix it, so you have my :+1:  for your code

File: mobile/android/android-emulator-services.ts [hunk 3]
```diff
@@ -224,19 +226,26 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
...
+					{ stdio: "ignore", detached: true });
 			} else {
-				this.$childProcess.spawn('emulator', ['-avd', image],
-					{ stdio: "ignore", detached: true }).unref();
+				emulatorProc = this.$childProcess.spawn('emulator', ['-avd', image],
+					{ stdio: "ignore", detached: true });
 			}
+			emulatorProc.on("error", () => emulatorProcEnded = true);
```
Comment (rosen-vladimirov) [inline mobile/android/android-emulator-services.ts]: You can log trace the error here

File: mobile/android/android-emulator-services.ts [hunk 4]
```diff
@@ -224,19 +226,26 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
...
+			emulatorProc.on("exit", () => emulatorProcEnded = true);
```
Comment (rosen-vladimirov) [inline mobile/android/android-emulator-services.ts]: I found an issue when starting native android emulator - we receive exit event during its start, so we exit the code :(

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Pull request closed without merging**